zilencer/views.py

```diff
@@ -116,20 +116,32 @@ def deactivate_remote_server(
 
 @csrf_exempt
 @require_post
-@has_request_variables
+@typed_endpoint
 def register_remote_server(
     request: HttpRequest,
-    zulip_org_id: str = REQ(str_validator=check_string_fixed_length(RemoteZulipServer.UUID_LENGTH)),
-    zulip_org_key: str = REQ(
-        str_validator=check_string_fixed_length(RemoteZulipServer.API_KEY_LENGTH)
-    ),
-    hostname: str = REQ(str_validator=check_capped_string(RemoteZulipServer.HOSTNAME_MAX_LENGTH)),
-    contact_email: str = REQ(),
-    new_org_key: Optional[str] = REQ(
-        str_validator=check_string_fixed_length(RemoteZulipServer.API_KEY_LENGTH), default=None
-    ),
+    *,
+    zulip_org_id: Annotated[
+        str,
+        StringConstraints(
```

Comment:
The `StringConstraints` feature does not provide support for strict length checking. As a result, a work-around has been implemented to address this issue.

Reply:
reason why pattern way falling, 
https://docs.pydantic.dev/2.0/migration/#patterns-regex-on-strings

Reply:
I might add Before/After validation for this too, instead of giving min/max length